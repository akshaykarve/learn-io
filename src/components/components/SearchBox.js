import React from 'react';
import '../ComponentStyle.css';

const SearchBox =({text, onSearchPlatform})=>{
	return (
		<div className='pa'>
			<input 
				className='searchBoxStyle'
				type='search'
                value={text}
				placeholder='Search platform'
				onChange={(e)=>{onSearchPlatform(e.target.value)}}
			/>
		</div>
	);
}

export default SearchBox;