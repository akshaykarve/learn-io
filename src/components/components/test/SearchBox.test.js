import React from 'react';
import renderer from 'react-test-renderer';
import {default as SearchBox} from "../SearchBox";

describe("Search Box Component", ()=>{
	it('no text', () => {
		
		const tree = renderer.create(
			<SearchBox text={''}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test text', () => {
		
		const tree = renderer.create(
			<SearchBox text={'test'}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});