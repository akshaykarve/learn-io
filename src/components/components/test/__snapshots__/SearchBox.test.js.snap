// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box Component no text 1`] = `
<div
  className="pa"
>
  <input
    className="searchBoxStyle"
    onChange={[Function]}
    placeholder="Search platform"
    type="search"
    value=""
  />
</div>
`;

exports[`Search Box Component test text 1`] = `
<div
  className="pa"
>
  <input
    className="searchBoxStyle"
    onChange={[Function]}
    placeholder="Search platform"
    type="search"
    value="test"
  />
</div>
`;
